subsys/storage/flash_map/flash_map_default.c

```diff
@@ -11,10 +11,10 @@
 #include <zephyr/zephyr.h>
 #include <zephyr/storage/flash_map.h>
 
-#define FLASH_AREA_FOO(part)						\
-	{.fa_id = DT_FIXED_PARTITION_ID(part),				\
-	 .fa_off = DT_REG_ADDR(part),					\
-	 .fa_dev = DEVICE_DT_GET(DT_MTD_FROM_FIXED_PARTITION(part)),	\
+#define FLASH_AREA_FOO(part)							\
+	{.fa_id = DT_FIXED_PARTITION_ID(part),					\
+	 .fa_off = DT_REG_ADDR(part),						\
+	 .fa_dev = DEVICE_DT_GET_OR_NULL(DT_MTD_FROM_FIXED_PARTITION(part)),	\
```

Comment:
I'm wondering on what is bad in build failure there on missing device driver?

After all that means that something is wrong with the project configuration: a partition miss a driver.  I don't think that it would be better if this will be converted to the run-time failure.

Reply:
@JordanYates Maybe I don understand the cases "when fixed-partition's exist on a flash device that has no driver compiled into the application".

Reply:
The use case that triggered the problem for me was building mcuboot on a board with an external flash chip. The bootloader needs no device drivers at all, but the board having partitions on external flash was breaking the build.

Reply:
OK. It used to be not an issue before, as device name was used here.

Reply:
> OK. It used to be not an issue before, as device name was used here.

It was, it would just appear in run-time. That is what the -ENODEV is returned for, from `flash_area_open`.

Reply:
@de-nordic @JordanYates  I assumed that mcuboot didn't use that external flash at all.

Reply:
@JordanYates @nvlsianpu Do we even need to generate entry for device-less partition?
The partition will not be accessible anyway and any attempt to use it will require attaching a device.
Accessing non-existent partition, IMHO, is programmers error and should be detected at tests, before device gets deployed; there is no difference in error processing, for me, whether we return -ENODEV because there is no device or -ENOENT because due to lack of device there is no partition defined.

Reply:
I agree that it makes no difference whether we generate the partition or not in terms of the errors generated, so I guess for ROM reasons we should just not generate it. This still doesn't fix the other issue in the PR description though.